File: SortingandSearching/01_distinctElemen/01_distinctElement.cpp
#include<bits/stdc++.h>
using namespace std;

int returnDistinctElement(vector<int>&arr,int n)
{
    sort(arr.begin(),arr.end());
    for(int i=1;i<n;i++)
        if(arr[i]==arr[i-1])
            return arr[i];
    return -1;
}

int main()
{
    int n = 5;
    vector<int>arr =  {2, 3, 2, 2, 3};
    cout<<returnDistinctElement(arr,n);

    return 0;
}

File: SortingandSearching/01_distinctElemen/01_distinctElement.java
import java.util.ArrayList;

class DistinctElement
{
    static int returnDistinctElement(ArrayList<Integer>arr,int n)
    {
        for(int i=1;i<n;++i)
            if(arr.get(i)==arr.get(i-1))
                return arr.get(i);

        return -1;
    }
    public static void main(String[] args) 
    {
        int n = 5;
        ArrayList<Integer>arr = new ArrayList<>(n);
        arr.add(2);
        arr.add(3);
        arr.add(2);
        arr.add(2);
        arr.add(3);
        System.out.println(returnDistinctElement(arr, n));
        
    }
}
